pulp_ansible/app/serializers.py

```diff
@@ -648,11 +811,51 @@ class CollectionVersionSerializer(ContentChecksumSerializer, CollectionVersionUp
 
     creating = True
 
+    def create(self, validated_data):
+        artifact = validated_data.get("artifact")
+        repository = validated_data.get("repository")
+        sigstore_verifying_services = repository.sigstore_verifying_service.all()
+
+        cv_instance = super().create(validated_data)
+
+        artifact_name = artifact.file.name
+        artifact_file = storage.open(artifact_name)
+
+        with tempfile.TemporaryDirectory() as tempdir:
+            with tarfile.open(fileobj=artifact_file, mode="r") as tar:
+                tar.extractall(path=tempdir)
+
+                for root, dirs, files in os.walk(tempdir):
+                    if ".ansible-sign" in dirs:
+                        manifest_path = Path(tempdir) / ".ansible-sign" / "sha256sum.txt"
+                        signature_path = Path(tempdir) / ".ansible-sign" / "sha256sum.txt.sigstore"
+                        if os.path.exists(manifest_path) and os.path.exists(signature_path):
+                            sigstore_bundle = Bundle().from_json(signature_path.read_bytes())
+
+                            with manifest_path.open("rb", buffering=0) as io:
+
+                                for sigstore_verifying_service in sigstore_verifying_services:
+                                    verification_result = sigstore_verifying_service.sigstore_verify(manifest=io, sigstore_bundle=sigstore_bundle)
```

Comment:
If none of the verifications pass, should the collection be rejected?

Reply:
I intentionally made it so that the collection can get uploaded even without any valid signature, as it seems to me like it should not be a criteria for admitting content. If you think this should be mandated on a repository configured with at least one Sigstore verifying service, let me know.